.github/workflows/build-images-action.yml

```diff
@@ -44,3 +44,13 @@ jobs:
             "BUILD_CONTAINER_IMAGE_GIT_REFERENCE": "${{ github.ref }}"
           }
         job_timeout: "1000"
+    - name: Slack Notification on Failure
+      if: ${{ failure() }}
+      uses: rtCamp/action-slack-notify@v2.0.2
+      env:
+        SLACK_TITLE: 'GitHub Action Failed in ${{ github.repository }}'
+        SLACK_COLOR: '#FF0000'
+        SLACK_MESSAGE: 'The GitHub Action workflow failed for baremetal operator image build.'
```

Comment:
So the default condition is  if: ${{ success() }}
Meaning that a step won't run if a failure has occurred in any earlier step

failure on the other hand will run in case any earlier job fails:

> Returns true when any previous step of a job fails. If you have a chain of dependent jobs, failure() returns true if any ancestor job fails.

So the if image build one fails the second image won't build but the notification will be sent.
Example this failed pipeline shows this: https://github.com/metal3-io/baremetal-operator/actions/runs/8108012548/job/22160469135